core/block/editor/lastused/lastused.go

```diff
@@ -0,0 +1,218 @@
+package lastused
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/anyproto/any-sync/app"
+	"github.com/anyproto/any-sync/app/logger"
+	"github.com/cheggaaa/mb/v3"
+	"github.com/gogo/protobuf/types"
+	"go.uber.org/zap"
+
+	"github.com/anyproto/anytype-heart/core/block/editor/smartblock"
+	"github.com/anyproto/anytype-heart/core/domain"
+	"github.com/anyproto/anytype-heart/pkg/lib/bundle"
+	coresb "github.com/anyproto/anytype-heart/pkg/lib/core/smartblock"
+	"github.com/anyproto/anytype-heart/pkg/lib/localstore/addr"
+	"github.com/anyproto/anytype-heart/pkg/lib/localstore/objectstore"
+	"github.com/anyproto/anytype-heart/space"
+	"github.com/anyproto/anytype-heart/space/clientspace"
+	"github.com/anyproto/anytype-heart/util/pbtypes"
+)
+
+const (
+	CName = "object-usage-updater"
+
+	maxInstallationTime = 5 * time.Minute
+	updateInterval      = 5 * time.Second
+)
+
+type Key interface {
+	URL() string
+	String() string
+}
+
+type message struct {
+	spaceId string
+	key     Key
+	time    int64
+}
+
+var log = logger.NewNamed("update-last-used-date")
+
+type ObjectUsageUpdater interface {
+	app.ComponentRunnable
+
+	UpdateLastUsedDate(spaceId string, key Key, timeStamp int64)
+}
+
+func New() ObjectUsageUpdater {
+	return &updater{}
+}
+
+type updater struct {
+	store        objectstore.ObjectStore
+	spaceService space.Service
+
+	ctx     context.Context
+	cancel  context.CancelFunc
+	started bool
+
+	msgBatch *mb.MB[message]
+}
+
+func (u *updater) Name() string {
+	return CName
+}
+
+func (u *updater) Init(a *app.App) error {
+	u.store = app.MustComponent[objectstore.ObjectStore](a)
+	u.spaceService = app.MustComponent[space.Service](a)
+	u.msgBatch = mb.New[message](0)
+	return nil
+}
+
+func (u *updater) Run(context.Context) error {
+	u.started = true
+	u.ctx, u.cancel = context.WithCancel(context.Background())
+	go u.lastUsedUpdateHandler()
+	return nil
+}
+
+func (u *updater) Close(context.Context) error {
+	if u.started {
+		u.cancel()
```

Comment:
`if u.cancel != nil {`

and `started` seems redundant

Reply:
Done!